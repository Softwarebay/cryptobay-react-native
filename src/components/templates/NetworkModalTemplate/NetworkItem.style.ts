import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  networkListItem: {
    paddingHorizontal: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  contentWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  dot: {
    width: 12,
    height: 12,
    borderRadius: 12,
    marginRight: 18,
  },
});
