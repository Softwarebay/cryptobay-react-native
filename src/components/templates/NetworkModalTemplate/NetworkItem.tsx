import React from 'react';
import { View, ViewStyle } from 'react-native';
import { Label } from '../../atoms';
import { styles } from './NetworkItem.style';

type NetworkItemProps = {
  networkName: string;
  networkColor: string;
  style?: ViewStyle | ViewStyle[];
};

export const NetworkItem: React.FC<NetworkItemProps> = ({
  networkName,
  networkColor,
  style,
}: NetworkItemProps) => {
  const dotColorStyle = { backgroundColor: networkColor } as ViewStyle;

  return (
    <View style={style}>
      <View style={styles.contentWrapper}>
        <View style={[styles.dot, dotColorStyle]} />
        <Label variant="secondary" value={networkName} />
      </View>

      {/* Icon */}
    </View>
  );
};
