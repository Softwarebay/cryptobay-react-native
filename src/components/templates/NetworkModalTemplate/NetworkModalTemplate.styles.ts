import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  networkModalTemplate: {
    width: '100%',
    height: '100%',
    justifyContent: 'flex-end',
  },
  blur: {
    height: '40%',
    padding: 8,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  indicator: {
    width: 40,
    height: 4,
    borderRadius: 4,
    backgroundColor: '#ABAFC4',
  },
  networkContent: {
    backgroundColor: '#17171A',
    height: '60%',
    padding: 24,
  },
  networkModalLabel: {
    marginBottom: 32,
    alignSelf: 'center',
    color: '#FFFFFF',
  },
  selectedNetworkItem: {
    marginBottom: 32,
    paddingLeft: 10,
  },
  networkListWrapper: {
    width: '100%',
  },
  networkListLabel: {
    marginBottom: 24,
    color: '#ABAFC4',
  },
  networkList: {
    marginBottom: 40,
    paddingLeft: 10,
  },
  networkListItem: {
    marginBottom: 18,
  },
  closeButton: {
    marginBottom: 54,
    alignSelf: 'flex-end',
  },
});
