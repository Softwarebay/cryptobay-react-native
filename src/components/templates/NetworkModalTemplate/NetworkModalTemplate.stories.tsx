import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';
import { NetworkModalTemplate } from './NetworkModalTemplate';

storiesOf('Templates/NetworkTemplate', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Network Template', () => (
    <NetworkModalTemplate onPressClose={action('onPressClose')} />
  ));
