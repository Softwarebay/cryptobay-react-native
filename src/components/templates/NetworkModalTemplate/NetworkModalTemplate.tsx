import { BlurView } from '@react-native-community/blur';
import React from 'react';
import { View, Pressable, FlatList } from 'react-native';

import { Label } from '../../atoms';
import { Button } from '../../molecules';

import { NetworkItem } from './NetworkItem';
import { styles } from './NetworkModalTemplate.styles';

const networksListMock = [
  { name: 'Ropsten Test', color: '#EA3943' },
  { name: 'Kovan Test', color: '#45F0D1' },
  { name: 'Goerli Test', color: '#5F97FF' },
];

type NetworkModalTemplateProps = {
  onPressClose: () => void;
};

export const NetworkModalTemplate: React.FC<NetworkModalTemplateProps> = ({
  onPressClose,
}: NetworkModalTemplateProps) => {
  return (
    <Pressable style={styles.networkModalTemplate} onPress={onPressClose}>
      <BlurView style={styles.blur}>
        <View style={styles.indicator} />
      </BlurView>

      <View style={styles.networkContent}>
        <Label style={styles.networkModalLabel} value="Network" />

        <NetworkItem
          style={styles.selectedNetworkItem}
          networkName="Ethereum Main"
          networkColor="#FEBF32"
        />

        <View style={styles.networkListWrapper}>
          <Label style={styles.networkListLabel} value="Other Networks" />
          <FlatList
            style={styles.networkList}
            data={networksListMock}
            renderItem={({ item }) => (
              <NetworkItem
                style={styles.networkListItem}
                networkName={item.name}
                networkColor={item.color}
              />
            )}
          />
        </View>

        <Button
          style={styles.closeButton}
          label="Close"
          variant="ghost"
          onPress={onPressClose}
        />
      </View>
    </Pressable>
  );
};
