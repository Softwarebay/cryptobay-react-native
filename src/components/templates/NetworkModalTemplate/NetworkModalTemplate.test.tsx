import React from 'react';
import '@testing-library/jest-native';
import 'react-native';
import renderer from 'react-test-renderer';

import { NetworkModalTemplate } from './NetworkModalTemplate';

describe('NetworkModalTemplate Template', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<NetworkModalTemplate onPressClose={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
