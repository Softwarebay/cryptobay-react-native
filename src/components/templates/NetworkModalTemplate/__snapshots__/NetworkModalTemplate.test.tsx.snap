// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkModalTemplate Template Component Snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "justifyContent": "flex-end",
      "width": "100%",
    }
  }
>
  <BlurView
    blurAmount={10}
    blurType="dark"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "height": "40%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "flex-end",
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ABAFC4",
            "borderRadius": 4,
            "height": 4,
            "width": 40,
          }
        }
      />
    </View>
  </BlurView>
  <View
    style={
      Object {
        "backgroundColor": "#17171A",
        "height": "60%",
        "padding": 24,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "marginBottom": 32,
          },
        ]
      }
    >
      Network
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 32,
            "paddingLeft": 10,
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 10,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 12,
                "height": 12,
                "marginRight": 18,
                "width": 12,
              },
              Object {
                "backgroundColor": "#FEBF32",
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins",
              },
              Object {
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 24,
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Ethereum Main
        </Text>
      </View>
      <View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          testID="svg-icon-check"
          vbHeight={24}
          vbWidth={24}
          width={24}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGCircle
              cx="12"
              cy="12.0001"
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              r="9.00375"
              stroke={
                Object {
                  "payload": 4285981288,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="1.5"
            />
            <RNSVGPath
              d="M8.4425 12.3392L10.6104 14.5071L10.5964 14.4931L15.4874 9.60205"
              propList={
                Array [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                Object {
                  "payload": 4285981288,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="1.5"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#ABAFC4",
              "marginBottom": 24,
            },
          ]
        }
      >
        Other Networks
      </Text>
      <RCTScrollView
        data={
          Array [
            Object {
              "color": "#EA3943",
              "name": "Ropsten Test",
            },
            Object {
              "color": "#45F0D1",
              "name": "Kovan Test",
            },
            Object {
              "color": "#5F97FF",
              "name": "Goerli Test",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "marginBottom": 70,
            "paddingLeft": 10,
          }
        }
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Array [
                  Object {
                    "marginBottom": 18,
                  },
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 12,
                        "height": 12,
                        "marginRight": 18,
                        "width": 12,
                      },
                      Object {
                        "backgroundColor": "#EA3943",
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Ropsten Test
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Array [
                  Object {
                    "marginBottom": 18,
                  },
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 12,
                        "height": 12,
                        "marginRight": 18,
                        "width": 12,
                      },
                      Object {
                        "backgroundColor": "#45F0D1",
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Kovan Test
                </Text>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Array [
                  Object {
                    "marginBottom": 18,
                  },
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingHorizontal": 10,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 12,
                        "height": 12,
                        "marginRight": 18,
                        "width": 12,
                      },
                      Object {
                        "backgroundColor": "#5F97FF",
                      },
                    ]
                  }
                />
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Poppins",
                      },
                      Object {
                        "fontSize": 14,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                    ]
                  }
                >
                  Goerli Test
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-end",
          "backgroundColor": "transparent",
          "borderRadius": 8,
          "color": "#FEBF32",
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 16,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins",
            },
            Object {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            Object {
              "color": "#FEBF32",
            },
          ]
        }
      >
        Close
      </Text>
    </View>
  </View>
</View>
`;
